app/src/main/java/org/jellyfin/androidtv/ui/search/SearchRepository.kt

```diff
@@ -0,0 +1,43 @@
+package org.jellyfin.androidtv.ui.search
+
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import org.jellyfin.androidtv.auth.repository.UserRepository
+import org.jellyfin.androidtv.util.apiclient.AsyncState
+import org.jellyfin.sdk.api.client.ApiClient
+import org.jellyfin.sdk.api.client.extensions.itemsApi
+import org.jellyfin.sdk.model.api.BaseItemKind
+import org.jellyfin.sdk.model.api.ItemFields
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+object SearchRepository: KoinComponent {
+
+	private const val QUERY_LIMIT = 25
+
+	private val apiClient: ApiClient by inject()
+
+	suspend fun search(
```

Comment:
Just have this return `Result<BaseItemDtoQueryResult>` instead of the AsyncState. Introducing that to our repositories might make sense but would need to be a separate pull request.

Reply:
No problem.  At some point I would like to add a loading spinner somewhere.  Will handle that elsewhere.

Reply:
Updated